### PR TITLE
Update to xUnit 2.0.0

### PR DESCRIPTION
**Pull request by suwatch**

### CODE REVIEW

**Comment by suwatch** [comment]
This is only update from Xunit 1.9 to Xunit 2.0.  I want to do just this first and make sure we are in par with the old Xunit 1.9 run.   Next step (different check in) is to turn on parallel runs.

<b>Note</b>:  Xunit VS runner is nuget based only (vsix deprecated).  In order to see in test explorer, you <b>must</b> first remove the Xunit VS runner from VS extensions (VS -> Tools -> Extension and Updates...).

**Comment by amitapl** [comment]
Looks good

File: Kudu.Core.Test/Infrastructure/XmlUtilityFacts.cs [hunk 1]
```diff
@@ -20,15 +19,17 @@ public void XmlUtilityReturnsValidCharacterStringsUnchanged(string input)
             Assert.Equal(input, output);
         }
 
-        [Theory]
-        [PropertyData("InvalidCharacterSet")]
-        public void XmlUtilityRemovesInvalidCharacters(string input)
+        [Fact]
```
Comment (pranavkm) [inline Kudu.Core.Test/Infrastructure/XmlUtilityFacts.cs]: Why change this?

File: Kudu.TestHarness/Xunit/KuduXunitTestCase.cs [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                if (TestMethodArguments != null)
+                {
+                    // Work around test arg issue
+                    Write(stream, DisplayName);
+                    // Write(stream, SerializationHelper.Serialize(TestMethodArguments));
+                }
+
+                stream.Position = 0;
+
+                var hash = new System.Security.Cryptography.SHA1Managed().ComputeHash(stream);
```
Comment (pranavkm) [inline Kudu.TestHarness/Xunit/KuduXunitTestCase.cs]: Dispose this?

**Comment by pranavkm** [comment]
:shipit:

File: Kudu.Core.Test/Infrastructure/XmlUtilityFacts.cs (hunk 1 above)
Comment (suwatch) [inline Kudu.Core.Test/Infrastructure/XmlUtilityFacts.cs]: this is a regression from xunit 1.9 when saving the result.  It tries to put that input value to XML result and choke due to invalid Xml character.

https://github.com/xunit/xunit/issues/380

File: Kudu.TestHarness/Xunit/KuduXunitTestCase.cs (hunk 2 above)
Comment (suwatch) [inline Kudu.TestHarness/Xunit/KuduXunitTestCase.cs]: definitely.

**Pull request merged**